### PR TITLE
Add agent.{id,ephemeral_id} to all beat events

### PR DESCRIPTION
**Pull request by andrewvc**
This adds the ECS fields `agent.id` and `agent.ephemeral_id` to all beats.

Heartbeat in particular needs these to execute aggregations queries in dashboards.

I could use some feedback here on the best place to add tests. AFAICT existing similar functionality for these baseline fields is not tested. Thoughts @urso ?

### CODE REVIEW

File: libbeat/beat/info.go [hunk 1]
```diff
@@ -21,12 +21,13 @@ import "github.com/gofrs/uuid"
...
-	Name        string    // configured beat name
-	Hostname    string    // hostname
-	UUID        uuid.UUID // ID assigned to beat instance
+	Beat          string    // The actual beat's name
+	IndexPrefix   string    // The beat's index prefix in Elasticsearch.
+	Version       string    // The beat version. Defaults to the libbeat version when an implementation does not set a version
+	Name          string    // configured beat name
+	Hostname      string    // hostname
+	UUID          uuid.UUID // ID assigned to beat instance
+	EphemeralUUID uuid.UUID // ID assigned to beat instance
```
Comment (ruflin) [inline libbeat/beat/info.go]: We already have a ephemeral id: https://github.com/elastic/beats/blob/master/libbeat/cmd/instance/metrics_common.go#L31

Yes, not easy to find. I think here is a better place for it.

Comment (andrewvc) [reply libbeat/beat/info.go]: Errr, this isn't a new UUID, we pull it from https://github.com/elastic/beats/pull/9404/files#diff-73bd591578392eb2d281f4778ac9a47eR88

Comment (andrewvc) [reply libbeat/beat/info.go]: Oops, I meant to say: https://github.com/elastic/beats/pull/9404/files#diff-f7226280e7734d6e3d88e7dde232602eR217

**Comment by andrewvc** [comment]
@urso yeah, I'm glad to rename `UUID` -> `EphemeralID`

File: libbeat/beat/info.go (hunk 1 above)
Comment (andrewvc) [reply libbeat/beat/info.go]: Actually, it already is pulled from where you mentioned (this line is the type declaration)

**Comment by andrewvc** [comment]
@urso @ruflin I've done the rename. That wound up touching a number of files FYI

**Comment by andrewvc** [comment]
Hit a flaky test. This PR is now blocked on the `SkipTest` PR for that here: https://github.com/elastic/beats/pull/9503

**Comment by andrewvc** [comment]
@ruflin changelog entry added. Let me know if we're LGTM (though we'll still be blocked on the Flaky test merge)

**Comment by ruflin** [comment]
@andrewvc The tests that are failing are golden files which do not seem to be up-to-date. But I could not reproduce it locally. Any chance you could rebase on master and see if it still happens?

**Comment by andrewvc** [comment]
@ruflin I ran those commands but it only updated some of the stuff in c246f83 . Still getting test failures: https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/2507/beat=filebeat,label=ubuntu/

**Comment by ruflin** [comment]
The problem is that the agent_id and ephemeral id change every time. So we should skip these two fields. Let me fix it locally and provide you with the code.

**Comment by ruflin** [comment]
Can you try to replace line https://github.com/elastic/beats/blob/master/filebeat/tests/system/test_modules.py#L189 with the following content?
```
    # These keys are host dependent
    host_keys = ["host.name", "agent.hostname", "agent.type", "agent.ephemeral_id", "agent.id"]
```

This exclude the two ids from being generated. Afterwards you need to run the command again.

**Comment by andrewvc** [comment]
Hmmm, looks like adding to the clean list wasn't enough to get a passing suite. Let's connect monday. Regenerating the fixtures didn't fix it either.

**Comment by ruflin** [comment]
I did run the branch locally and on my end it removes the ids also from the files which still contain it in this PR. Not sure what happened? Can you run it again?

**Comment by andrewvc** [comment]
@ruflin the tests were failing for me at the last line your command list (where `nosetests` is run). Weirdly, after trying a few things, the cause seemed to be line breaks in the formatting of the new `geo.name` field's description. I removed them in https://github.com/elastic/beats/pull/9404/commits/dd19ce00dadf69c6f8487a82e954fa162d628ad1 . WFG to see if that did it.

**Comment by andrewvc** [comment]
Hmmm, still getting errors on filebeat. Looks like it's still looking for agent.id? 

https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/2605/

**Comment by ruflin** [comment]
@andrewvc Yes, some files still have agent.id inside: https://github.com/elastic/beats/pull/9404/files#diff-adbf1bb07ce8f5f874891602efc0f5e7R4

**Comment by andrewvc** [comment]
jenkins, please retest this

**Comment by andrewvc** [comment]
OK , just rebased with a `make update` (again).

Anything else needed here @ruflin?

FYI, this PR has a high chance of needing extra rebases due to it being sensitive to any field changes in any other beat.

File: libbeat/_meta/fields.ecs.yml [hunk 2]
```diff
@@ -230,10 +230,8 @@
             type: keyword
             description: >
               User-defined description of a location, at the level of granularity they care about.
-
```
Comment (ruflin) [inline libbeat/_meta/fields.ecs.yml]: Spotting this only now. Removing these newlines changed all the above docs which I don't think was intentional?

**Comment by ruflin** [comment]
@andrewvc Can you rebase this one?

@kuisathaverat One interesting thing I saw in this PR: It seems the Check step in the Jenkinsfile fails here because of a merge conflict. Does this mean what the Jenkinsfile tests is the PR merged with master?

**Comment by kuisathaverat** [comment]
>Does this mean what the Jenkinsfile tests is the PR merged with master?

yes, the default configuration makes a merge before building the PR.

**Comment by andrewvc** [comment]
Failure unrelated

**Comment by andrewvc** [comment]
@ruflin @urso do y'all think this should be backported at all? 7.0 does seem like a fine time to introduce this.

**Comment by ruflin** [comment]
I don't see a need to backport this at the moment.

**Comment by urso** [comment]
No need to backport.

We found this change to break indexing if `%{[beat.version]}` was used. Will need to investigate why the code string formatter didn't produce an error here.

**Pull request merged**